### PR TITLE
use route hooks from the pagewrapper

### LINKED ISSUE #1121: Projects details view is not routed properly
**Describe the bug**
When we try to view any project (eg. https://zubhub.unstructured.studio/projects/0848243c-5381-45f3-bacb-e60c58fdda73) the request is redirected to https://zubhub.unstructured.studio/projects/undefined

**To Reproduce**
Go to Zubhub, list projects, then click on any project to view its detail. Address bar shows full route, https://zubhub.unstructured.studio/projects/0848243c-5381-45f3-bacb-e60c58fdda73 but the request is actually sent to https://zubhub.unstructured.studio/projects/undefined which returns 404

**Expected behavior**
We see project properly

**Screenshots**
image

### PR DESCRIPTION
**Pull request by coderatomy**
fixes https://github.com/unstructuredstudio/zubhub/issues/1121
fixes validation errors

### CODE REVIEW

**Comment by coderatomy** [comment]
more work has to be done since `params` is extensively used https://github.com/search?q=repo%3Aunstructuredstudio%2Fzubhub%20props.params&type=code

File: zubhub_frontend/zubhub/src/views/create_project/script.js [hunk 1]
```diff
@@ -693,21 +693,8 @@ export const validationSchema = Yup.object().shape({
...
-                value.split(',').forEach(material => {
-                    if (material) {
-                        is_empty = false;
-                    }
-                });
-
-            return !is_empty;
-        }),
-    category: Yup.string(),
+    materials_used: Yup.array().required('required'),
```
Comment (NdibeRaymond) [inline zubhub_frontend/zubhub/src/views/create_project/script.js]: No idea why we keep removing  validations instead of getting them to work properly

Comment (NdibeRaymond) [reply zubhub_frontend/zubhub/src/views/create_project/script.js]: If we are going to work on this then we should fix it completely

**Review by NdibeRaymond — CHANGES_REQUESTED** [review]

**Comment by NdibeRaymond** [comment]
> fixes #1121 fixes validation errors

* fixed  eslint  errors
*  improvements to App.js  and PageWrapper.jsx
* removed materials_used and category field validation issues. Those require their own issue and PR

**Pull request merged**